WordPress/wordcamp.org
# Add organizer debrief survey

Closes #930

This PR implements what has been concluded in the ticket. The logic refers to #997.

**Flow**

- Manually published Organizer Survey (event debrief) in Central.

- When the plugin is activated:
  - Create a `tix_email` in draft status.
    -  The email will contain a token generated by hashing the access token key and WordCamp ID, to prevent the survey page from being accessed and abused by individuals other than the lead organizers.
  - Schedule a `daily` cron job.

- When the cron job runs:
  - Check if we already sent an email for this camp.
  - Check to make sure the camp ended exactly 2 days ago.
  - Add the lead organizer as an attendee and associate it with `tix_email`.
  - Add `tix_email` to the queue (set to `pending`)
    - _email gets picked up by camptix cron email sending queue `tix_scheduled_every_ten_minutes`_

## Screenshots

| View | Screenshot |
| --- | --- |
| Email queue | <img width="1525" alt="image" src="https://github.com/WordPress/wordcamp.org/assets/18050944/36c72cc4-7e74-46ea-9a28-446c525a6864"> |
| Manually published Organizer Survey (event debrief) page in central | <img width="1528" alt="image" src="https://github.com/WordPress/wordcamp.org/assets/18050944/28bd62d4-5cf8-47c9-b990-9a8f7a4e9572"> |
| Draft Survey Email | <img width="1234" alt="image" src="https://github.com/WordPress/wordcamp.org/assets/18050944/75fc0963-b93f-46bb-a18a-f0e19ec4a05a"> |
| Survey Email Sent | <img width="608" alt="image" src="https://github.com/WordPress/wordcamp.org/assets/18050944/3d3d3165-8183-458e-ad78-85508545ea17"> |
| Survey | <img width="821" alt="image" src="https://github.com/WordPress/wordcamp.org/assets/18050944/abbe8b65-d240-4789-bb6e-b1a5bb3c4c53"> |

## How to test

1. Create a Next-Gen Event WordCamp site and fill in `Organizer Name` and `Email Address` inside the Organizing Team.
   - Make the Start Date in the past.
   - Make the End Date the current date.
2. Visit Cenral
   - Go to Pages
     - Manually publish `Organizer Survey (event debrief)`
     - View the content, which should be the same as the description in ticket.
2. Visit Next-Gen
   - Go to Ticket -> E-Mail
     - Expect to see an email `Organizer Survey (event debrief)` in `Draft` mode.
     - The email content is pending an update from the Tardis Team. 
   - Run the following wp cli script to make sure the cron jobs are scheduled
     - `wp cron event list --url=https://events.wordpress.test/{city}/{year}/{type}` 
     - expect to see the following cron job: `wc_organizer_debrief_survey_email`
   - Go back to the Next-Gen Wordcamp Post type and update the End Date to exact 2 days ago.
   - Run this script to set WordCamp to closed `wp post update {post_id} --post_status=wcpt-closed --url=https://central.wordcamp.test/`
   - Run the following cli command:
     - run `wp cron event run wc_organizer_debrief_survey_email --url=https://events.wordpress.test/{city}/{year}/{type}` 
       - Should succeed
       - Status of `Organizer Survey (event debrief)` in Ticket -> E-Mail should be `pending`.
       - An attendee should be added in Ticket -> Attendees with `tix_first_name` and `tix_email` the same as the one you input when creating the event.
     - run `wp cron event run tix_scheduled_every_ten_minutes -url=https://events.wordpress.test/{city}/{year}/{type}`
       - `Organizer Survey (event debrief)` in Ticket -> E-Mail should be published.
       - In, http://localhost:1080/, there should be an email with Subject `Organizer Survey (event debrief)`. 
         - Visit the survey link on the front end and submit.
 4. Make sure none of this exists on the traditional WordCamp.
